be/src/exec/spill/spill_components.cpp

```diff
@@ -184,24 +181,22 @@ Status PartitionedSpillerWriter::reset_partition(const std::vector<const SpillPa
         _max_level = std::max(_max_level, partition->level);
         _partition_set[partitions[i]->partition_id] = true;
     }
-    RETURN_IF_ERROR(_prepare_partitions(_runtime_state));
-    return Status::OK();
+    _prepare_partitions(_runtime_state);
 }
 
-Status PartitionedSpillerWriter::reset_partition(RuntimeState* state, size_t num_partitions) {
+void PartitionedSpillerWriter::reset_partition(RuntimeState* state, size_t num_partitions) {
     num_partitions = BitUtil::next_power_of_two(num_partitions);
     num_partitions = std::min<size_t>(num_partitions, 1 << max_partition_level);
     num_partitions = std::max<size_t>(num_partitions, _spiller->options().init_partition_nums);
 
     _level_to_partitions.clear();
     _id_to_partitions.clear();
     std::fill(_partition_set.begin(), _partition_set.end(), false);
-    RETURN_IF_ERROR(_init_with_partition_nums(state, num_partitions));
-    RETURN_IF_ERROR(_prepare_partitions(state));
-    return Status::OK();
+    _init_with_partition_nums(state, num_partitions);
+    _prepare_partitions(state);
 }
 
-Status PartitionedSpillerWriter::_init_with_partition_nums(RuntimeState* state, int num_partitions) {
+void PartitionedSpillerWriter::_init_with_partition_nums(RuntimeState* state, int num_partitions) {
     DCHECK((num_partitions & (num_partitions - 1)) == 0);
     DCHECK(num_partitions > 0);
     DCHECK(_level_to_partitions.empty());
```

Comment:
what if these `DCHECK` is rewriteen to `return error status`?

Reply:
> what if these `DCHECK` is rewriteen to `return error status`?

These are some basic assumptions and constraints. If they are not met, it must be an internal implementation bug. I think adding these DCHECK to verify it during test is enough, not necessary to return status.